code/Model/SiteTree.php

```diff
@@ -1187,6 +1188,14 @@ public function canView($member = null)
             return true;
         }
 
+        // check for specific users
+        if ($this->CanViewType === InheritedPermissions::ONLY_THESE_MEMBERS
+            && $member
+            && \in_array($member->ID, $this->ViewerMembers()->column('ID'), true)
```

Comment:
```suggestion
            && $this->ViewerMembers()->filter('ID', $member->ID)->count() > 0
```
Slightly more efficient, both the db query itself and not having to iterate over IDs. It won't be significant in 99% of cases, but I can imagine someone adding _all_ of the members from a large organisation into this field at some point.